scripts/packaging/createupmpackages.ps1

```diff
@@ -8,33 +8,38 @@
 .PARAMETER OutputDirectory
     Where should we place the output? Defaults to ".\artifacts"
 .PARAMETER PackageVersion
-    What version of the artifacts should we build? If unspecified, the highest
-    git tag pointing to HEAD is searched. If none is found, an error is reported.
+    What version of the artifacts should we build?
 .PARAMETER ProjectRoot
     The root folder of the project.
 #>
 param(
     [ValidatePattern("^\d+\.\d+\.\d+")]
     [string]$NodejsVersion = "12.18.0",
     [string]$OutputDirectory = ".\artifacts\upm",
-    [ValidatePattern("^\d+\.\d+\.\d+-?[a-zA-Z0-9\.]*$")] # todo - format of d.d.d[-preview.0-9.0-9]
+    [ValidatePattern("^\d+\.\d+\.\d+-?[a-zA-Z0-9\.]*$")]
     [string]$PackageVersion,
     [string]$ProjectRoot
 )
 
+$startPath = "$(Get-Location)"
+
+if (-not $ProjectRoot) {
+    # ProjectRoot was not specified, presume the current location is Root\scripts\packaging
+    $ProjectRoot = Resolve-Path "$startPath\..\.." 

```

Comment:
Is this used for the build machine or for local runs?

In general I'm still averse to using CWD based things because they are not reliable.

Reply:
it is for both. the build machine passes in the project root. the local runs can omit it

Reply:
Local runs would still assume that it's being run in a certain folder - still a weird dependency to have.

If you wanted to base this instead off the the location of the script itself (so that it works regardless of what folder you're in), that could be okay.

This would prevent you from using a script in a DIFFERENT repo from affecting the CWD version, but  Idon't know how often you want to do that anyway

Reply:
let me see what i can do 😄

Reply:
removed that section. ProjectRoot is now required to be passed on the command line